pandas/core/apply.py

```diff
@@ -131,10 +131,18 @@ def get_result(self):
 
         # ufunc
         elif isinstance(self.f, np.ufunc):
+            results = {}
             with np.errstate(all='ignore'):
-                results = self.f(self.values)
+                for col in self.columns:
+                    if is_sparse(self.obj.dtypes[col]):
+                        fill = self.f(self.obj.dtypes[col].fill_value)
+                        result = self.f(self.obj[col])
+                        result = result.to_sparse(fill_value=fill)
+                    else:
+                        result = self.f(self.obj[col])
+                    results[col] = result
             return self.obj._constructor(data=results, index=self.index,
-                                         columns=self.columns, copy=False)
```

Comment:
You probably need to pass `columns` here to ensure that the columns are in the correct order. Add a test where the columns aren't sorted (like `['b', 'a', 'c']`) and verify that it fails on py35 and older.